lookup/client.go

```diff
@@ -119,6 +120,11 @@ func (c *Client) SiteID(ctx context.Context) (string, error) {
 // If the endpoint is found, its TLS certificate is also added to the vim25.Client's trusted host thumbprints.
 // If the Lookup Service is not available, the given path is returned as the default.
 func EndpointURL(ctx context.Context, c *vim25.Client, path string, filter *types.LookupServiceRegistrationFilter) string {
+	// Services running on vCenter can bypass lookup service.
+	if useSidecar := internal.UsingEnvoySidecar(c); useSidecar {
+		return fmt.Sprintf("http://%s%s", c.URL().Host, path)
```

Comment:
Not sure I follow. Wouldn't this mean we always return the 'base' address of the Envoy sidecar rather than the service specific route? (which is only known around line 131 I think)

(It seems like it would be easier to verify with a test)

Reply:
sorry, didnt understand. 
path passed above to the function, for suppose for licensing service is /ls/sdk (https://gitlab.eng.vmware.com/core-build/govmomi-internal/-/blob/master/ls/client.go#L37) , we are appending the same to 'base' address .

L131, in this case returns , https://sc1-10-78-227-166.eng.vmware.com:443/ls/sdk (from the logs)

> (It seems like it would be easier to verify with a test)
you meant to add unit test or any other way?

Reply:
Ah, misunderstood. So callers can pass in the 'default' path and have it returned if the Envoy sidecar is in use _or_ if lookup service is broken after this change.

Reply:
yes, if lookup service is broken and not using envoy sidecar, it still follows the original flow and just return the path

Reply:
Okay, makes sense, thanks for clarifying.